deltacat/compute/compactor_v2/utils/task_options.py

```diff
@@ -298,8 +301,9 @@ def merge_resource_options_provider(
 
     total_memory = total_memory * (1 + TOTAL_MEMORY_BUFFER_PERCENTAGE / 100.0)
     debug_memory_params["total_memory_with_buffer"] = total_memory
-    logger.debug(
-        f"[Merge task {index}]: Params used for calculating merge memory: {debug_memory_params}"
-    )
+    if memory_logs_enabled:
+        logger.debug(
+            f"[Merge task {index}]: Params used for calculating merge memory: {debug_memory_params}"
+        )
```

Comment:
Please create a higher order function on logger that can take in `do_print` and calls logger if required.